lightning/src/offers/parse.rs

```diff
@@ -193,6 +193,8 @@ pub enum Bolt12SemanticError {
 	UnexpectedPaymentHash,
 	/// A signature was expected but was missing.
 	MissingSignature,
+	/// There is insufficient liquidity to complete the payment.
+	InsufficientLiquidity,
```

Comment:
I'm a little hesitant adding another variant as it's not really a semantic error. Admittedly, we've sort of stretched the use of `Bolt12SemanticError` to be used with the builder functions (e.g., using `MissingPaths` when we can't *create* a reply path) instead of creating an error type that wraps this to support other variants. Do other reviewer have any opinions on this? (@tnull @TheBlueMatt)

Reply:
We already have `Bolt12ParseError` which has a variant that wraps `Bolt12SemanticError`.

Reply:
Is there an existing alternative error type that would be better for this case?

Reply:
The only thing close would be `InsufficientAmount`, but using it would also be stretching its intended use.

Reply:
Discussed offline with @tnull. Let's use a new error type in the `ChannelManager` functions (maybe `Bolt12CreationError`?) that is an enum with one variant wrapping `Bolt12SemanticError` and other variants for errors that can occur outside of the builder (e.g., failing to create a blinded path, not enough liquidity, duplicate payment id, etc.). See `Bolt12ParseError` for an example. This may require removing variants like `DuplicatePaymentId` from `Bolt12SemanticError`.